insights/parsers/crictl_ps.py

```diff
@@ -0,0 +1,201 @@
+
+        self.data = []
+
+        # Skip the header line
+        for line in content[1:]:
+            line = line.strip()
+            if not line:
+                continue
+
+            # Parse each line manually to handle the space-containing "created" field
+            parsed = self._parse_line(line)
+            if parsed:
+                self.data.append(parsed)
+
+    def _parse_line(self, line):
+        """
+        Parse a single line of crictl ps output.
+
+        This method handles the complex parsing logic required for crictl output
+        where the "created" field may contain spaces and the image field can be
+        very long.
+
+        Args:
+            line (str): A single line from crictl ps output.
+
+        Returns:
+            dict: Parsed container record with all fields, or None if parsing fails.
+
+        Note:
+            The parsing logic:
+            1. Splits the line by whitespace
+            2. First two fields are always container_id and image
+            3. Finds the state field by looking for known state keywords
+            4. Everything between image and state is the "created" field
+            5. Remaining fields are mapped in order: name, attempt, pod_id, pod
+        """
+        # Split by whitespace
+        parts = line.split()
+
+        if len(parts) < 8:
+            return None
+
+        # The structure is:
+        # container_id image created_parts... state name attempt pod_id pod
+
+        # First two parts are always container_id and image
+        container_id = parts[0]
+        image = parts[1]
+
+        # Find the state field (it's always a single word like "Running", "Exited", etc.)
+        # Look for common state values
+        state_keywords = ['Running', 'Exited', 'Created', 'Unknown', 'ContainerCreating', 'Pending']
+        state_index = None
+        for i, part in enumerate(parts[2:], 2):
+            if part in state_keywords:
+                state_index = i
+                break
+
+        if state_index is None:
+            return None
+
+        # Everything between image and state is the "created" field
+        created_parts = parts[2:state_index]
+        created = ' '.join(created_parts)
+
+        # The remaining parts after state
+        remaining = parts[state_index + 1:]
+
+        if len(remaining) < 4:
+            return None
+
+        # The last 4 parts are: name, attempt, pod_id, pod
+        name = remaining[-4]
+        attempt = remaining[-3]
+        pod_id = remaining[-2]
+        pod = remaining[-1]
+
+        # The state is at state_index
+        state = parts[state_index]
+
+        return {
+            'container_id': container_id,
+            'image': image,
+            'created': created,
+            'state': state,
+            'name': name,
+            'attempt': attempt,
+            'pod_id': pod_id,
+            'pod': pod
```

Comment:
Nitpick: 
it would be good if save these local variables e.g.
```python
            'container_id': parts[0],  # First two parts are always container_id and image
            'image': parts[1],
            'created': ' '.join(parts[2:state_index]),  # Everything between image and state is the "created" field
            'state': parts[state_index],  # The state is at state_index
            'name': remaining[-4],  # The last 4 parts are: name, attempt, pod_id, pod
            'attempt': remaining[-3],
            'pod_id': remaining[-2],
            'pod': remaining[-1]
```